tools/generator/src/Logger.swift

```diff
@@ -22,33 +22,66 @@ final class StdoutOutputStream: TextOutputStream {
     }
 }
 
+enum TerminalColor: Int {
+    case red = 31
+    case green = 32
+    case yellow = 33
+    case magenta = 35
+
+    func colorize(_ input: String) -> String {
+        return "\u{001B}[\(self.rawValue);1m\(input)\u{001B}[0m"
+    }
+}
+
 /// The logger that is used when not running tests.
 final class DefaultLogger<E: TextOutputStream, O: TextOutputStream>: Logger {
     private var standardError: E
     private var standardOutput: O
+    private var colorize: Bool
 
-    init(standardError: E, standardOutput: O) {
+    func disableColors() {
+        self.colorize = false
+    }
+
+    init(standardError: E, standardOutput: O, colorize: Bool) {
         self.standardError = standardError
         self.standardOutput = standardOutput
+        self.colorize = colorize
+    }
+
+    func format(color: TerminalColor, prefix: String, message: String) -> String {
+        if self.colorize {
+            return "\(color.colorize("\(prefix):")) \(message)"
+        } else {
+            return "\(prefix): \(message)"
+        }
     }
 
-    func logDebug(_ message: @autoclosure () -> String) {
-        // TODO: Colorize
-        print("DEBUG: \(message())", to: &self.standardOutput)
+    func logDebug(_ message: String) {
+        print(
+            self.format(color: .yellow, prefix: "DEBUG", message: message),
+            to: &self.standardOutput
+        )
     }
 
-    func logInfo(_ message: @autoclosure () -> String) {
-        // TODO: Colorize
-        print("INFO: \(message())", to: &self.standardOutput)
+    func logInfo(_ message: String) {
+        print(
+            self.format(color: .green, prefix: "INFO", message: message),
+            to: &self.standardOutput
+        )
     }
 
-    func logWarning(_ message: @autoclosure () -> String) {
-        // TODO: Colorize
-        print("WARNING: \(message())", to: &self.standardError)
+    func logWarning(_ message: String) {
+        print(
+            self.format(color: .magenta, prefix: "WARNING", message: message),
+            to: &self.standardError
+        )
     }
 
-    func logError(_ message: @autoclosure () -> String) {
-        // TODO: Colorize
-        print("ERROR: \(message())", to: &self.standardError)
+    func logError(_ message: String) {
+        print(
+            self.format(color: .red, prefix: "ERROR", message: message),
```

Comment:
Just so it's still tracked: we want this to be bold as well.

Reply:
It _is_ bold, all colors are bold now. It's the `;1` ANSI code.